core/arch/arm/mm/core_mmu.c

```diff
@@ -314,32 +502,33 @@ bool core_mmu_mattr_is_ok(uint32_t mattr)
  */
 bool core_pbuf_is(uint32_t attr, tee_paddr_t pbuf, size_t len)
 {
-	struct map_area *map;
+	struct tee_mmap_region *map;
 
 	/* Empty buffers complies with anything */
 	if (len == 0)
 		return true;
 
 	switch (attr) {
 	case CORE_MEM_SEC:
-		return ((platform_pbuf_is_t)bootcfg_pbuf_is)(attr, pbuf, len);
+		return pbuf_is(attr, pbuf, len);
```

Comment:
`pbuf_is()` is also doing switch-case on `attr` so I'm moving that code into this function instead.